test/framework/src/main/java/org/elasticsearch/cli/CommandTestCase.java

```diff
@@ -56,6 +61,7 @@ protected CommandTestCase(boolean spaceInPath) {
 
     @Before
     public void resetTerminal() throws IOException {
+        capturedLogLevel = LoggerFactory.provider().getRootLevel();
```

Comment:
We've bitten ourselves before by tests doing this and then all subsequent tests create insane amounts of log output causing all kinds of havoc. Is there any reason not to do this in `ESTestCase` so that we catch all these scenarios?

Reply:
Seems reasonable. I pushed [2a9144a](https://github.com/elastic/elasticsearch/pull/123742/commits/2a9144a1017b9e19f296cc21571ee1d6359a0643)

Reply:
I had to relax this a little bit. For some reason, capturing and restoring per-test in ESTestCase causes several tests which test logging to fail, seemingly because they don't get the expected log levels. So I changed this to be at the suite level for now.